File: BOJ_Swift/그리디/BOJ_2212_센서.swift
let n = Int(readLine()!)!
let k = Int(readLine()!)!

var sensor = readLine()!.split(separator: " ").compactMap { Int($0)! }
sensor.sort()

var li = [Int]()

for i in 0..<n-1 {
    li.append(sensor[i+1] - sensor[i])
}

var res = 0
li.sort()

if n > k {
    for i in 0..<n-k {
        res += li[i]
    }
}

print(res)

File: BOJ_Python/그리디/BOJ_2212_센서.py
import sys

n = int(sys.stdin.readline())
k = int(sys.stdin.readline())
sensor = list(map(int, sys.stdin.readline().split()))
sensor.sort()

li = []

for i in range(0, n-1):
    li.append(sensor[i+1]-sensor[i])

res = 0
li.sort()

for i in range(n-k):
    res += li[i]

print(res)
